include/nbl/ext/ImGui/ImGui.h

```diff
@@ -3,68 +3,47 @@
-		void SetNextItemWidth(float nextItemWidth);
-		void SetWindowSize(float width, float height);
-		void Text(char const* label, ...);
-		void InputFloat(char const* label, float* value);
-		void InputFloat2(char const* label, float* value);
-		void InputFloat3(char const* label, float* value);
-		void InputFloat4(char const* label, float* value);
-		void InputFloat3(char const* label, nbl::core::vector3df& value);
-		bool Combo(char const* label, int32_t* selectedItemIndex, char const** items, int32_t itemsCount);
-		bool Combo(const char* label, int* selectedItemIndex, std::vector<std::string>& values);
-		void SliderInt(char const* label, int* value, int minValue, int maxValue);
-		void SliderFloat(char const* label, float* value, float minValue, float maxValue);
-		void Checkbox(char const* label, bool* value);
-		void Spacing();
-		void Button(char const* label, std::function<void()> const& onPress);
-		void InputText(char const* label, std::string& outValue);
-		[[nodiscard]] bool HasFocus();		
-		[[nodiscard]] bool IsItemActive();
-		[[nodiscard]] bool TreeNode(char const* name);
-		void TreePop();
+		bool render(nbl::video::IGPUCommandBuffer* commandBuffer, const uint32_t frameIndex);
+		void update(float deltaTimeInSec, float mousePosX, float mousePosY, size_t mouseEventsCount, ui::SMouseEvent const * mouseEvents); // TODO: Keyboard events
+		int registerListener(std::function<void()> const& listener);
+		bool unregisterListener(int listenerId);
+		
+		void* getContext();
+		void setContext(void* imguiContext);
 
 	private:
-
-		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> CreateDescriptorSetLayout();
-
-		void CreatePipeline(video::IGPURenderpass* renderpass, video::IGPUPipelineCache* pipelineCache);
-		void CreateFontTexture(video::IGPUCommandBuffer* cmdBuffer, video::IQueue* queue);
-		void UpdateDescriptorSets();
+		core::smart_refctd_ptr<video::IGPUDescriptorSetLayout> createDescriptorSetLayout();
+		void createPipeline(video::IGPURenderpass* renderpass, video::IGPUPipelineCache* pipelineCache);
+		void createFontTexture(video::IGPUCommandBuffer* cmdBuffer, video::IQueue* queue);
+		void updateDescriptorSets();
 		void createSystem();
-		void CreateFontSampler();
-		void CreateDescriptorPool();
-		void HandleMouseEvents(float mousePosX, float mousePosY, size_t mouseEventsCount, ui::SMouseEvent const * mouseEvents) const;
+		void createFontSampler();
+		void createDescriptorPool();
+		void handleMouseEvents(float mousePosX, float mousePosY, size_t mouseEventsCount, ui::SMouseEvent const * mouseEvents) const;
 
 		core::smart_refctd_ptr<system::ISystem> system;
 		core::smart_refctd_ptr<system::ILogger> logger;
 		core::smart_refctd_ptr<video::IUtilities> utilities;
-
 		core::smart_refctd_ptr<video::ILogicalDevice> m_device;
 		core::smart_refctd_ptr<video::IGPUSampler> m_fontSampler;
 		core::smart_refctd_ptr<video::IDescriptorPool> m_descriptorPool;
 		core::smart_refctd_ptr<video::IGPUDescriptorSet> m_gpuDescriptorSet;
```

Comment:
these are super weird to belong to ImGUI

Reply:
resolved